### PR TITLE
Add support of running general Ansible modules on EosHost

### PR DESCRIPTION
**Pull request by wangxin**
### Description of PR

Summary:
Fixes # (issue)
This is a re-submit of PR #1721 which was reverted because of an issue.

* Add support of running general Ansible modules on EosHost

The EOS hosts support two types of command line interface:
1. Linux shell
2. EOS shell

This change is to add the support of running general Ansible modules
that can only be executed under Linux shell.

* If not able to get the explicitly defined username and
password, fallback to use the value of "fanout_admin_user"
and "fanout_admin_password". It is unlikely to break the
existing code.

### Type of change

- [ ] Bug fix
- [x] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

#### What is the motivation for this PR?
This change is to add the support of running general Ansible modules that can only be executed under Linux shell.

#### How did you do it?
Improved the EosHost class defined in tests/common/devices.py to make it accepts two sets of credential:

The first set of credential exposes the EOS CLI interface. When use this set of credential, we use ansible_connection network_cli.
The second set of credential exposes the linux shell of EOS. When use this set of credential, we use ansible_connection ssh.
If name of the ansible module to be executed on EOS hosts starts with "eos_", then use ansible_connection network_cli to run the module. Otherwise, use ansible_connection ssh to run the module.

#### How did you verify/test it?
Test run the test scripts that depend on EosHost.

#### Any platform specific information?
Only EoSHost is enhanced. The OnyxHost may need to be enhanced in the similar way.

### CODE REVIEW

File: tests/common/devices.py [hunk 1]
```diff
@@ -604,54 +604,83 @@ class EosHost(AnsibleHostBase):
...
+            gather_facts (bool, optional): Whether to gather some basic facts. Defaults to False.
+        '''
+        self.eos_user = eos_user
+        self.eos_passwd = eos_passwd
+        self.shell_user = shell_user
+        self.shell_passwd = shell_passwd
+        AnsibleHostBase.__init__(self, ansible_adhoc, hostname)
         self.localhost = ansible_adhoc(inventory='localhost', connection='local', host_pattern="localhost")["localhost"]
 
+    def __getattr__(self, item):
```
Comment (lguohan) [inline tests/common/devices.py:626]: if you can rename item -> module_name, it will be more readable.

**Comment by lguohan** [comment]
retest this please

**Comment by lguohan** [comment]
retest this please

File: tests/common/devices.py (hunk 1 above)
Comment (wangxin) [reply tests/common/devices.py]: Updated.

**Comment by lguohan** [comment]
retest this please

**Comment by lguohan** [comment]
retest this please

**Comment by neethajohn** [comment]
retest this please

**Comment by lguohan** [comment]
retest this please

**Pull request merged**